GenesToPhenotypes.ipynb

```diff
@@ -107,14 +114,155 @@
+    "                                        'Project' : project_name,\n",
+    "                                        'Dataset': dataset_name})\n",
+    "                            # we have the start of a row now\n",
+    "                            # but we want to print out as many rows\n",
+    "                            # as there are ontology mappings\n",
+    "                            # so if there is mapping to 1 ontology term\n",
+    "                            # print 1 row, if there are 2 ontology terms\n",
+    "                            # print 2 rows etc\n",
+    "                            numberOfRows = len(ontologies)/6\n",
+    "                            # this is 3 pairs of ontology values per\n",
+    "                            # mapping, add the ontology mappings and print\n",
+    "                            n = 1\n",
+    "                            while (n <= numberOfRows):\n",
+    "                                row.update({ontologies[0]: ontologies[1],\n",
+    "                                            ontologies[2]: ontologies[3],\n",
+    "                                            ontologies[4]: ontologies[5]})\n",
+    "                                # remove that set of ontology mappings\n",
+    "                                ontologies = ontologies[6:]\n",
+    "                                writer.writerow(row)\n",
+    "                                n = n + 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Load data "
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 6,
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "csvfile = NamedTemporaryFile(\"w\", delete=False, newline='')\n",
+    "try:\n",
+    "    fieldnames = [\n",
+    "        'Gene', 'Screen', 'Plate', 'Project', 'Dataset', 'Image',\n",
+    "        'Phenotype', 'Phenotype Term Name', 'Phenotype Term Accession',\n",
+    "        'Phenotype Term Accession URL']\n",
+    "    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)\n",
+    "    writer.writeheader()\n",
+    "\n",
+    "    for gene in genes:\n",
+    "        qs1 = {'key': TYPE, 'value': gene}\n",
+    "        url1 = SCREENS_PROJECTS_URL.format(**qs1)\n",
+    "        json = session.get(url1).json()\n",
+    "        for s in json['screens']:\n",
+    "            qs2 = {'key': TYPE, 'value': gene, 'screen_id': s['id']}\n",
+    "            url2 = PLATES_URL.format(**qs2)\n",
+    "            parse_annotation(writer, session.get(url2).json(), s['name'], 'plates')\n",
+    "        for p in json['projects']:\n",
+    "            qs2 = {'key': TYPE, 'value': gene, 'project_id': p['id']}\n",
+    "            url2 = DATASETS_URL.format(**qs2)\n",
+    "            parse_annotation(writer, session.get(url2).json(), p['name'], 'datasets')           \n",
+    "finally:\n",
+    "    csvfile.close()"
```

Comment:
What is our answer to the question about "how do I get this csv file locally now ?". 
Is thee an easy way to enable download of this file directly from the Jupyter notebook ?
I think we do not need to offer this if too hard or complex to implement, just to know the answer.
I understand that the cell below will give me the output into the stdout of the notebook if I change the number of the displayed lines to very high number, I will have it, but a clean csv would be something nice too.

Reply:
It's currently written to a temporary file:
```py
csvfile = NamedTemporaryFile("w", delete=False, newline='')
```
If instead you wrote it to a file in the current directory and don't delete it, it should be downloadable via the jupyter file list

Reply:
We could